client/src/main/scala/org/http4s/client/middleware/Retry.scala

```diff
@@ -162,8 +162,11 @@ object RetryPolicy {
     isErrorOrRetriableStatus(result)
 
   def isErrorOrRetriableStatus[F[_]](result: Either[Throwable, Response[F]]): Boolean =
+    isErrorOrStatus(result, RetriableStatuses)
+
+  def isErrorOrStatus[F[_]](result: Either[Throwable, Response[F]], status: Set[Status]): Boolean =
```

Comment:
If we could get a comment on this vs the other use case. As these were the only option previously. A little documentation on how and why you might use it would be useful as it was only clear to me after I looked at the unit test, which isn't in the same place as the source.

Reply:
@ChristopherDavenport  I have added a description. Let me know if it makes sense or if there is a better wording for it